akka-sample-distributed-workers-scala/src/main/scala/worker/WorkManager.scala

```diff
@@ -1,129 +1,101 @@
+import akka.Done
+import akka.actor.typed.delivery.WorkPullingProducerController.{MessageWithConfirmation, RequestNext}
+import akka.actor.typed.delivery.{ConsumerController, WorkPullingProducerController}
+import akka.actor.typed.receptionist.ServiceKey
 import akka.actor.typed.scaladsl.Behaviors
-import akka.actor.typed.ActorRef
-import akka.actor.typed.receptionist.{ Receptionist, ServiceKey }
-import akka.cluster.pubsub.{ DistributedPubSub, DistributedPubSubMediator }
-import akka.persistence.typed.PersistenceId
-import akka.persistence.typed.scaladsl.Effect
-import akka.persistence.typed.scaladsl.EventSourcedBehavior
-import worker.WorkState.WorkAccepted
-import worker.WorkState.WorkDomainEvent
-import akka.actor.typed.scaladsl.adapter._
-import worker.WorkState.WorkCompleted
-import worker.WorkState.WorkStarted
-import worker.WorkState.WorkerFailed
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.persistence.typed.{PersistenceId, RecoveryCompleted}
+import akka.persistence.typed.scaladsl.{Effect, EventSourcedBehavior}
+import akka.util.Timeout
+import worker.WorkState.{WorkAccepted, WorkCompleted, WorkDomainEvent, WorkStarted}
 
-import scala.collection.mutable.ListBuffer
-import scala.concurrent.duration.{ Deadline, FiniteDuration, _ }
+import scala.concurrent.duration.{FiniteDuration, _}
+import scala.util.{Failure, Success}
 
 /**
  * The work manager actor keep tracks of all available workers, and all scheduled and ongoing work items
  */
 object WorkManager {
 
+  val ManagerServiceKey = ServiceKey[ConsumerController.Command[WorkerCommand]]("worker-service-key")
+
   val WorkerServiceKey: ServiceKey[WorkerCommand] = ServiceKey[WorkerCommand]("workerService")
   val ResultsTopic = "results"
 
   final case class Ack(workId: String) extends CborSerializable
 
-  sealed trait WorkerStatus
-  case object Idle extends WorkerStatus
-  final case class Busy(workId: String, deadline: Deadline) extends WorkerStatus
-  final case class WorkerState(ref: ActorRef[WorkerCommand], status: WorkerStatus)
-
-  sealed trait Command extends CborSerializable
-  private final case class UpdatedWorkers(workers: Receptionist.Listing) extends Command
-
-  // Messages from Workers
-  final case class WorkerRequestsWork(workerId: String, replyTo: ActorRef[WorkerCommand]) extends Command
-  final case class WorkIsDone(workerId: String, workId: String, result: Any, replyTo: ActorRef[WorkerCommand])
-      extends Command
-  final case class WorkFailed(worker: ActorRef[WorkerCommand], workId: String) extends Command
-
   // Responses to requests from workers
   sealed trait WorkerCommand
   final case class DoWork(work: Work) extends WorkerCommand
-  final case class WorkAck(id: String) extends WorkerCommand
-  case object WorkAvailable extends WorkerCommand
 
-  // External commands
+  sealed trait Command extends CborSerializable
   final case class SubmitWork(work: Work, replyTo: ActorRef[WorkManager.Ack]) extends Command
```

Comment:
`job: Any` - that shouldn't be possible to serialize with Jackson
need some more concrete type, maybe String is enough for the sample